uwrt_mars_rover_drivetrain/uwrt_mars_rover_drivetrain_hw/src/uwrt_mars_rover_drivetrain_hw_real.cpp

```diff
@@ -28,20 +28,31 @@ void UWRTMarsRoverDrivetrainHWReal::read(const ros::Time & /*time*/, const ros::
   // TODO: change to use tpdos instead of queries
   for (const auto &joint_name : joint_names_) {
     // position measured in Radians from starting position
-    actuator_joint_states_[joint_name].actuator_position =
-        REVOLUTIONS_TO_RADIANS_FACTOR *
-        motor_controller_->readAbsoluteEncoderCount(roboteq_actuator_index_[joint_name]) /
-        ticks_per_revolution_[joint_name];
+    try {
+      actuator_joint_states_[joint_name].actuator_position =
+          REVOLUTIONS_TO_RADIANS_FACTOR *
+          motor_controller_->readAbsoluteEncoderCount(roboteq_actuator_index_[joint_name]) /
+          ticks_per_revolution_[joint_name];
+    } catch (const std::exception &e) {
+      // TODO: replace roboteq exceptions with ros diagnostics!! #106
+      ROS_ERROR_STREAM_NAMED(name_, "CAUGHT EXCEPTION: " << e.what());
+    }
```

Comment:
ew exceptions. is there some code we didnt write that throws exceptions?

Reply:
third party ros code, as well as the base roscpp code have exceptions. The C++ STL also does throw exceptions. Exceptions in C++ are not inherently a bad thing.

However, in this case here, they're slated for replacing with ros diagnostics.